File: src/main/java/tafqeetj/numbers/QuantitativeNumberNamesFactory.java
package tafqeetj.numbers;

public abstract class QuantitativeNumberNamesFactory {
    private final static QuantitiveNumberNames thousands;
    private final static QuantitiveNumberNames millions;
    private final static QuantitiveNumberNames billions;
    private final static QuantitiveNumberNames trillions;

    static {
        thousands = new QuantitiveNumberNames(
                "ألف",
                "ألفان",
                "آلاف",
                "ألفًا"
        );
        millions = new QuantitiveNumberNames(
                "مليون",
                "مليونان",
                "ملايين",
                "مليونًا"
        );
        billions = new QuantitiveNumberNames(
                "مليار",
                "ملياران",
                "مليارات",
                "مليارًا"
        );
        trillions = new QuantitiveNumberNames(
                "ترليون",
                "ترليونان",
                "ترليونات",
                "ترليونًا"
        );
    }

    public QuantitiveNumberNames getThousands() {
        return thousands;
    }

    public QuantitiveNumberNames getMillions() {
        return millions;
    }

    public QuantitiveNumberNames getBillions() {
        return billions;
    }

    public QuantitiveNumberNames getTrillions() {
        return trillions;
    }
}

File: src/main/kotlin/tech/moussa/tafqeetk/numbers/QuantitativeNumberNamesFactory.kt
package tech.moussa.tafqeetk.numbers

abstract class QuantitativeNumberNamesFactory {
    // TODO: Localization for different languages
    private val thousands = QuantitativeNumberNames(
        "ألف",
        "ألفين",
        "آلاف",
        "ألفًا"
    )
    private val millions = QuantitativeNumberNames(
        "مليون",
        "مليونين",
        "ملايين",
        "مليونًا"
    )
    private val billions = QuantitativeNumberNames(
        "مليار",
        "مليارين",
        "مليارات",
        "مليارًا"
    )
    private val trillions = QuantitativeNumberNames(
        "ترليون",
        "ترليونين",
        "ترليونات",
        "ترليونًا"
    )
    fun getThousands(): QuantitativeNumberNames {
        return this.thousands
    }
    fun getMillions(): QuantitativeNumberNames {
        return this.millions
    }
    fun getBillions(): QuantitativeNumberNames {
        return this.billions
    }
    fun getTrillions(): QuantitativeNumberNames {
        return this.trillions
    }
}